controllers/agentcluster_controller.go

```diff
@@ -219,7 +219,7 @@ func (r *AgentClusterReconciler) getControlPlane(ctx context.Context, log logrus
 
 func (r *AgentClusterReconciler) findOrCreateClusterDeployment(ctx context.Context, log logrus.FieldLogger, agentCluster *capiproviderv1.AgentCluster) (ctrl.Result, error) {
 	clusterDeployment, err := r.findClusterDeployment(ctx, agentCluster)
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Alternatively you could do this with `client.IgnoreNotFound(err)` in `findClusterDeployment` assuming you always want a successful return in this case.

Reply:
Great idea! It's a bit cleaner too. Modified as suggested.